File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.py
def solution(food):
    result = []
    for i in range(1, len(food)):
        count = food[i] // 2
        for _ in range(count):
            result.append(str(i))
        
    result_reversed = sorted(result, reverse = True)
    
    answer = ''.join(result)
    answer2 = ''.join(result_reversed)
    
    return answer + '0' + answer2

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.java
class Solution {
    public String solution(int[] food) {
        String answerA = "";
        
        for(int i = 1; i < food.length; i++) {
            for(int j = 1; j <= food[i]/2; j++){
                answerA = answerA.concat(Integer.toString(i));
            }
        }
        
        StringBuilder sb = new StringBuilder(answerA);
        String answerB = sb.reverse().toString();
        
        String answer = answerA.concat("0").concat(answerB);
        
        return answer;
    }
}